### PR TITLE
Add informational popovers to app diff

### PR DESCRIPTION
**Pull request by proteusvacuum**
https://dimagi-dev.atlassian.net/browse/SLTNS-217

Adds popovers to the app diff view to give more information on each change.

Screenshot from 2019-08-05 14-55-22
Screenshot from 2019-08-05 14-55-11
Screenshot from 2019-08-05 14-54-48

FF: App Diffs

### CODE REVIEW

**Review by emord — APPROVED** [review]
been a long time since I've looked at js popovers, but looks reasonable.

Approved pending my questions about whether or not this change is backwards compatiable. (i think they aren't stored anywhere, but wanted to be sure)

File: corehq/apps/app_manager/app_schemas/form_metadata.py [hunk 1]
```diff
@@ -40,35 +40,46 @@
...
-    calculate = StringProperty(choices=DIFF_STATES)
-    relevant = StringProperty(choices=DIFF_STATES)
-    required = StringProperty(choices=DIFF_STATES)
-    comment = StringProperty(choices=DIFF_STATES)
-    setvalue = StringProperty(choices=DIFF_STATES)
-    constraint = StringProperty(choices=DIFF_STATES)
-    options = DictProperty()    # {option: state}
-    load_properties = DictProperty()  # {case_type: {property: state}}
-    save_properties = DictProperty()  # {case_type: {property: state}}
+    question = ObjectProperty(_Change)
```
[the remark this answers is not in the document]
Comment (proteusvacuum) [reply corehq/apps/app_manager/app_schemas/form_metadata.py]: They are not stored or cached.

File: corehq/apps/app_manager/app_schemas/form_metadata.py [hunk 2]
```diff
@@ -377,25 +395,44 @@ def _mark_case_properties(self, first_question, second_question, attribute):
...
+        item.changes[key] = _Change(type=ADDED, new_value=new_value)
+
+    def _mark_item_changed(self, first_item, second_item, key):
+        self._set_contains_changes(first_item)
+        self._set_contains_changes(second_item)
+        if self._is_translatable_property(first_item[key], second_item[key]):
+            change_class = _TranslationChange
+        else:
+            change_class = _Change
+        change = change_class(type=CHANGED, old_value=first_item[key], new_value=second_item[key])
```
[the remark this answers is not in the document]
Comment (proteusvacuum) [reply corehq/apps/app_manager/app_schemas/form_metadata.py]: no. the keys are predefined and are always present on any item that is passed into this method.

File: corehq/apps/app_manager/app_schemas/form_metadata.py [hunk 3]
```diff
@@ -377,25 +395,44 @@ def _mark_case_properties(self, first_question, second_question, attribute):
...
             self._set_contains_changes(first_question)
             self._set_contains_changes(second_question)
 
     def _mark_item_removed(self, item, key):
         self._set_contains_changes(item)
-        item.changes[key] = REMOVED
+        try:
+            old_value = item[key]
+        except KeyError:
+            old_value = None
```
[the remark this answers is not in the document]
Comment (proteusvacuum) [reply corehq/apps/app_manager/app_schemas/form_metadata.py]: `item` is a `JsonObject`, and I don't think JsonObject implements `get`

**Pull request merged**